infinispan-client-quickstart fails with  java.lang.NoClassDefFoundError: org/junit/rules/TestRule

infinispan-client-quickstart fails with  java.lang.NoClassDefFoundError: org/junit/rules/TestRule

`mvn clean verify -f infinispan-client-quickstart`

```
[INFO] Running org.acme.infinispan.client.InfinispanGreetingResourceTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.05 s <<< FAILURE! - in org.acme.infinispan.client.InfinispanGreetingResourceTest
[ERROR] org.acme.infinispan.client.InfinispanGreetingResourceTest.testHelloEndpoint  Time elapsed: 0.003 s  <<< ERROR!
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
Caused by: java.lang.reflect.InvocationTargetException
Caused by: java.lang.RuntimeException: Error waiting for test resource future to finish.
Caused by: java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: org/junit/rules/TestRule
Caused by: java.lang.NoClassDefFoundError: org/junit/rules/TestRule
Caused by: java.lang.ClassNotFoundException: org.junit.rules.TestRule

```

@karesti fyi

--- a/test-framework/infinispan-client/pom.xml
+++ b/test-framework/infinispan-client/pom.xml
@@ -66,12 +66,9 @@
                 </exclusion>
             </exclusions>
         </dependency>
-        <!-- needed to make the junit version converge -->
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <version>${junit4.version}</version>
-            <optional>true</optional>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit4-mock</artifactId>
         </dependency>
     </dependencies>
 

